packages/headless/src/lib/headless.service.ts

```diff
@@ -0,0 +1,531 @@
+    result: QueryObserverResult<T>,
+    listener: (result: FetchResult<T>) => void
+  ) =>
+    listener({
+      data: result.data,
+      error: result.error,
+      status: result.status,
+      isLoading: result.isLoading,
+      isFetching: result.isFetching,
+      isError: result.isError,
+    });
+
+  private callUpdateListener = <
+    TData = unknown,
+    TError = unknown,
+    TVariables = unknown
+  >(
+    result: MutationObserverResult<TData, TError, TVariables>,
+    listener: (result: UpdateResult<TData, TError, TVariables>) => void
+  ) =>
+    listener({
+      data: result.data,
+      error: result.error,
+      status: result.status,
+      isLoading: result.isLoading,
+      isError: result.isError,
+    });
+
+  public initializeSession({
+    listener,
+    onSuccess,
+    onError,
+  }: {
+    listener: (result: FetchResult<ISession>) => void;
+    onSuccess?: (session: ISession) => void;
+    onError?: (error: unknown) => void;
+  }) {
+    this.session = null;
+    this.queryClient.clear();
+
+    const { unsubscribe } = this.queryService.subscribeQuery({
+      options: {
+        ...this.sessionQueryOptions,
+        onSuccess: (session) => {
+          this.session = session;
+          this.applyToken(session.token);
+          this.initializeSocket(session.token);
+          onSuccess?.(session);
+        },
+        onError,
+      },
+      listener: (result) => this.callFetchListener(result, listener),
+    });
+
+    return unsubscribe;
+  }
+
+  public async fetchOrganization({
+    listener,
+    onSuccess,
+    onError,
+  }: {
+    listener: (result: FetchResult<IOrganizationEntity>) => void;
+    onSuccess?: (session: IOrganizationEntity) => void;
+    onError?: (error: unknown) => void;
+  }) {
+    this.assertSessionInitialized();
+
+    const { unsubscribe } = this.queryService.subscribeQuery({
+      options: {
+        ...this.organizationQueryOptions,
+        onSuccess,
+        onError,
+      },
+      listener: (result) => this.callFetchListener(result, listener),
+    });
+
+    return unsubscribe;
+  }
+
+  public fetchUnseenCount({
+    listener,
+    onSuccess,
+    onError,
+  }: {
+    listener: (result: FetchResult<{ count: number }>) => void;
+    onSuccess?: (data: { count: number }) => void;
+    onError?: (error: unknown) => void;
+  }) {
+    this.assertSessionInitialized();
+
+    const { unsubscribe } = this.queryService.subscribeQuery({
+      options: {
+        ...this.unseenCountQueryOptions,
+        onSuccess,
+        onError,
+      },
+      listener: (result) => this.callFetchListener(result, listener),
+    });
+
+    return unsubscribe;
+  }
+
+  public listenUnseenCountChange({
+    listener,
+  }: {
+    listener: (unseenCount: number) => void;
+  }) {
+    this.assertSessionInitialized();
+
+    if (this.socket) {
+      this.socket.on(
+        'unseen_count_changed',
+        (data?: { unseenCount: number }) => {
+          if (!isNaN(data?.unseenCount)) {
```

Comment:
What you were doing works as it will filter many wrong inputs (undefined, null, a function, etc) but I would discourage the use of global `isNaN` due its implementation.

I'd rather use `Number.isInteger` like:
```suggestion
          if (Number.isInteger(data?.unseenCount)) {
```

Reply:
The code for socket handling was moved from the NC package, haven't touched it much.
But I agree with what you're saying, will change it, it's even more readable like that ;)